### PR TITLE
Fancy lot layout 📱

### PR DESCRIPTION
**Pull request by ashfurrow**
Continued from #1516, work towards #1085.

This introduces a new collection view layout for live auctions. It looks like this: 

I've tested it and gotten reasonable performance on a 5S, there may be need to improve it further on a 4S. It also needs support for 4" and 3.5" screens, as well as disabling the fancy layout for iPad. I've added these as TODOs on #1085 so that we can review and merge this as-is.

### CODE REVIEW

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift [hunk 1]
```diff
@@ -233,19 +243,16 @@ extension HostScrollViewDelegate: UIScrollViewDelegate {
...
     // When we scroll programmatically with/out animation.
     func scrollViewDidEndScrollingAnimation(scrollView: UIScrollView) {
         lotImageCollectionView.setContentOffset(scrollView.contentOffset, animated: false)
-        lotImageCollectionView.reloadData()
     }
 
     // When the user has released their finger and the scroll view is sliding to a gentle stop.
     func scrollViewDidEndDecelerating(scrollView: UIScrollView) {
         lotImageCollectionView.setContentOffset(scrollView.contentOffset, animated: false)
-        lotImageCollectionView.reloadData()
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift]: this is great

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |246.1ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-lot-view/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
201.4ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-lot-view/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
171.0ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-lot-view/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift [hunk 2]
```diff
@@ -1,8 +1,24 @@
 import UIKit
 
+typealias RelativeIndex = Int
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift]: cute

File: Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+
+protocol LiveAuctionFancyLotCollectionViewDelegateLayout: class {
+    func aspectRatioForIndex(index: RelativeIndex) -> CGFloat
+    func thumbnailURLForIndex(index: RelativeIndex) -> NSURL
+}
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift]: protocol has to be owned by a class, interesting

File: Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift [hunk 4]
```diff
@@ -20,7 +36,218 @@ class LiveAuctionFancyLotCollectionViewLayout: UICollectionViewFlowLayout {
...
+            applyFancyLayoutToAttributes(&layoutAttributes, position: .NextOverflow)
+        case 0:
+            applyFancyLayoutToAttributes(&layoutAttributes, position: .Previous)
+        case 1:
+            applyFancyLayoutToAttributes(&layoutAttributes, position: .Current)
+        case 2 where ratioDragged < -0.5:
+            applyFancyLayoutToAttributes(&layoutAttributes, position: .PreviousUnderflow)
+        default: // case 2:
+            applyFancyLayoutToAttributes(&layoutAttributes, position: .Next)
+        }
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionFancyLotCollectionViewLayout.swift]: nitpick: use a `default: break` instead of skipping 2

**Comment by orta** [comment]
Alright, given it a full look through, looks good 👍 
great job, way better than my version and about the same amount of time.

**Comment by alloy** [comment]
Looking sharp!

**Pull request merged**